### PR TITLE
[Doc] Add some specification about the support for amd64/arm64

### PR DESCRIPTION
**Pull request by chaoyli**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/developers/build-starrocks/Build_in_docker.md [hunk 1]
```diff
@@ -2,15 +2,15 @@
 
 ## Image
 
-The dev environment image was named as `dev-env`, and was a CentOS7 based image, since branch-2.5, Ubuntu-22.04 is recommended over CentOS7 distribution, the image is renamed to `dev-env-ubuntu` accordingly.
+The name of the development environment images is of the form `dev-env-xxx`. If you use Ubuntu 22.04, you can download the image initiated with `dev-env-ubuntu`, or if you use CentOS 7, you can download the image initiated with `dev-env-centos`.
```
Comment (evelynzhaojie) [inline docs/developers/build-starrocks/Build_in_docker.md:5]: The name of the development environment images is in the form of `dev-env-xxx`. If you use Ubuntu 22.04, you can download the image initiated with `dev-env-ubuntu`. If you use CentOS 7, you can download the image initiated with `dev-env-centos`.

File: docs/developers/build-starrocks/Build_in_docker.md [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
+
+StarRocks support both AMD64-based Linux and ARM64-based Linux. The Docker daemon automatically pull the corresponding image based on the CPU you use.
```
Comment (evelynzhaojie) [inline docs/developers/build-starrocks/Build_in_docker.md:7]: 动词形式，supports, pulls

**Review by evelynzhaojie — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-2.5 branch-3.0

**Comment by EsoragotoSpirit** [comment]
@mergify backport branch-3.0

**Pull request merged**